chia/consensus/blockchain.py

```diff
@@ -940,10 +943,11 @@ async def get_block_generator(
                         raise ValueError(Err.GENERATOR_REF_HAS_NO_GENERATOR)
                     result.append(ref_block.transactions_generator)
                 else:
-                    [gen] = await self.block_store.get_generators_at([ref_height])
-                    if gen is None:
-                        raise ValueError(Err.GENERATOR_REF_HAS_NO_GENERATOR)
-                    result.append(gen)
+                    heights_from_db.append(ref_height)
+            refs_from_db = await self.block_store.get_generators_at(heights_from_db)
+            if refs_from_db is None or len(refs_from_db) != len(heights_from_db):
+                raise ValueError(Err.GENERATOR_REF_HAS_NO_GENERATOR)
+            result.extend(refs_from_db)
```

Comment:
I don't think this is correct. The block-references is a list, not a set. The order we pass those generators to the block generator has to match that order.

Reply:
I think it's fine to not implement this optimization. In practice, our blocks have 0 or 1 block reference anyway

Reply:
are they just sorted by height ?

Reply:
I don't believe there are any constraints on them. you may have duplicates. It's just a list.

if the question is how *our* farmer will create blocks, we just have 0 or 1 block reference, but that doesn't define the protocol